### PR TITLE
{bio}[foss/2021a] hivtrace v0.6.2 w/ Python 3.9.5

### PR DESCRIPTION
**Pull request by i-mtz**
(created using `eb --new-pr`)

### CODE REVIEW

**Comment by smoors** [comment]
@i-mtz welcome to EasyBuild and thank you for your contribution!

File: easybuild/easyconfigs/h/hivtrace/hivtrace-0.6.2-foss-2021a.eb [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+# This two files come bundled compressed but the software expects them uncompressed
+postinstallcmds = [
+    'gunzip -d  %(installdir)s/lib/python%(pyshortver)s/site-packages/hivtrace/rsrc/LANL.FASTA.gz',
+    'gunzip -d  %(installdir)s/lib/python%(pyshortver)s/site-packages/hivtrace/rsrc/LANL.TN93OUTPUT.csv.gz',
+]
+
+sanity_check_paths = {
+    'files': ['bin/hivtrace'],
+    'dirs': ['lib/python%(pyshortver)s/site-packages'],
+}
```
Comment (smoors) [inline easybuild/easyconfigs/h/hivtrace/hivtrace-0.6.2-foss-2021a.eb:70]: @i-mtz can you add a `sanity_check_commands` here, as a minimal check that it works? something like this (the command needs to return 0):
```
sanity_check_commands = ['hivtrace --help']
```

File: easybuild/easyconfigs/h/hivtrace/hivtrace-0.6.2-foss-2021a.eb [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+homepage = 'https://github.com/veg/hivtrace'
+description = """HIV-TRACE is an application that identifies potential transmission clusters within a
+ supplied FASTA file with an option to find potential links against the Los Alamos HIV Sequence Database."""
+
+toolchain = {'name': 'foss', 'version': '2021a'}
+
+dependencies = [
+    ('Python', '3.9.5'),
+    ('Biopython', '1.79'),
+    ('tn93', '1.0.7'),
```
Comment (smoors) [inline easybuild/easyconfigs/h/hivtrace/hivtrace-0.6.2-foss-2021a.eb:20]: @i-mtz can you add the easyconfig for this software?

File: easybuild/easyconfigs/h/hivtrace/hivtrace-0.6.2-foss-2021a.eb [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    ('Pysam', '0.16.0.1'),
+]
+
+use_pip = True
+sanity_pip_check = True
+
+exts_list = [
+    ('hyphy-python', '0.1.10', {
+        'modulename': 'HyPhy',
+        'source_urls': ['https://pypi.python.org/packages/source/h/hyphy-python'],
```
Comment (smoors) [inline easybuild/easyconfigs/h/hivtrace/hivtrace-0.6.2-foss-2021a.eb:31]: @i-mtz the source_urls can be removed for all the extensions, as pypi is the default for extensions.

File: easybuild/easyconfigs/h/hivtrace/hivtrace-0.6.2-foss-2021a.eb (hunk 2 above)
Comment (i-mtz) [reply easybuild/easyconfigs/h/hivtrace/hivtrace-0.6.2-foss-2021a.eb:20]: @smoors  Should I add it in this PR or as a separate one?

Comment (smoors) [reply easybuild/easyconfigs/h/hivtrace/hivtrace-0.6.2-foss-2021a.eb:20]: @i-mtz you can choose, both is ok, but it's probably easier to just add it here.
if you do it in a separate PR, make sure to mention it at the top of this PR.

File: easybuild/easyconfigs/t/tn93/tn93-1.0.7-GCCcore-10.3.0.eb [hunk 4]
```diff
@@ -0,0 +1,48 @@
+# This file is an EasyBuild reciPY as per https://github.com/hpcugent/easybuild
+# Author: Iñaki Mtz de Ilarduya Muñoz
+# sciCORE - University of Basel
+# SIB Swiss Institute of Bioinformatics
+
+easyblock = "CMakeMake"
+
+name = 'tn93'
```
Comment (smoors) [inline easybuild/easyconfigs/t/tn93/tn93-1.0.7-GCCcore-10.3.0.eb:8]: @i-mtz from the github page it looks like the official name is in capital letters: `TN93`

File: easybuild/easyconfigs/t/tn93/tn93-1.0.7-GCCcore-10.3.0.eb [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+version = '1.0.7'
+
+homepage = 'https://github.com/veg/tn93'
+description = """his is a simple program meant to compute pairwise distances between
```
Comment (smoors) [inline easybuild/easyconfigs/t/tn93/tn93-1.0.7-GCCcore-10.3.0.eb:12]: typo
```suggestion
description = """This is a simple program meant to compute pairwise distances between
```

File: easybuild/easyconfigs/t/tn93/tn93-1.0.7-GCCcore-10.3.0.eb [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+
+builddependencies = [
+    ('CMake', '3.20.1'),
+    ('binutils', '2.36.1'),
+]
+
+
+build_cmd = ' make '
+
+
```
Comment (smoors) [inline easybuild/easyconfigs/t/tn93/tn93-1.0.7-GCCcore-10.3.0.eb:35]: nit: remove second empty line

File: easybuild/easyconfigs/t/tn93/tn93-1.0.7-GCCcore-10.3.0.eb [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+# will allways end up with /usr/local as the install prefix
+install_cmd = 'mkdir -p %(installdir)s/bin &&'
+install_cmd += ' cp {fasta_diff,nucfreqsfasta,readreduce,selectreads,seqcoverage,'
+install_cmd += 'ShortestPathTN93,tn93,tn93-cluster,validate_fasta}'
+install_cmd += ' %(installdir)s/bin '
+
+sanity_check_paths = {
+    'files': ['bin/tn93'],
+    'dirs': [],
+}
```
Comment (smoors) [inline easybuild/easyconfigs/t/tn93/tn93-1.0.7-GCCcore-10.3.0.eb:46]: @i-mtz can you add a sanity check command here too?

**Review by smoors — CHANGES_REQUESTED** [review]
@i-mtz thanks for the updates, just a few more suggestions, I think we're almost there.

File: easybuild/easyconfigs/t/tn93/tn93-1.0.7-GCCcore-10.3.0.eb (hunk 7 above)
Comment (i-mtz) [reply easybuild/easyconfigs/t/tn93/tn93-1.0.7-GCCcore-10.3.0.eb:46]: I added the `tn93 -v` as `tn93 -h` uses `1` as exit status...

File: easybuild/easyconfigs/t/tn93/tn93-1.0.7-GCCcore-10.3.0.eb (hunk 4 above)
Comment (i-mtz) [reply easybuild/easyconfigs/t/tn93/tn93-1.0.7-GCCcore-10.3.0.eb:8]: I have renamed the name inside and also the easyconfig, however I'm not sure about how to remove the old tn93 easyconfig from the pr (besides from the GitHub interface). Is there a way to do it from the `eb --update-pr`?

Comment (smoors) [reply easybuild/easyconfigs/t/tn93/tn93-1.0.7-GCCcore-10.3.0.eb:8]: you can remove an easyconfig by prefixing with a colon:
```
eb --update-pr 14691 --pr-commit-msg <message> TN93-1.0.7-GCCcore-10.3.0.eb :tn93-1.0.7-GCCcore-10.3.0.eb
```

Comment (i-mtz) [reply easybuild/easyconfigs/t/tn93/tn93-1.0.7-GCCcore-10.3.0.eb]: Ah cool thanks for the tip, just updated the PR without the old lowercase easyconfig.

File: easybuild/easyconfigs/h/hivtrace/hivtrace-0.6.2-foss-2021a.eb [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+homepage = 'https://github.com/veg/hivtrace'
+description = """HIV-TRACE is an application that identifies potential transmission clusters within a
+ supplied FASTA file with an option to find potential links against the Los Alamos HIV Sequence Database."""
+
+toolchain = {'name': 'foss', 'version': '2021a'}
+
+dependencies = [
+    ('Python', '3.9.5'),
+    ('Biopython', '1.79'),
+    ('tn93', '1.0.7'),
```
Comment (smoors) [inline easybuild/easyconfigs/h/hivtrace/hivtrace-0.6.2-foss-2021a.eb:20]:
```suggestion
    ('TN93', '1.0.7'),
```

File: easybuild/easyconfigs/t/TN93/TN93-1.0.7-GCCcore-10.3.0.eb [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+
+toolchain = {'name': 'GCCcore', 'version': '10.3.0'}
+
+source_urls = ['https://github.com/veg/tn93/archive/refs/tags/']
+sources = ['v%(version)s.tar.gz']
+checksums = ['4a088689a23e8dc92d4484431ab0de2a0f5f05109d2a56e7bd55c0cc6459c4db']
+
+unpack_options = '--strip-components=1'
+
+separate_build_dir = True
```
Comment (smoors) [inline easybuild/easyconfigs/t/TN93/TN93-1.0.7-GCCcore-10.3.0.eb:23]: this is not needed - it's already the default

File: easybuild/easyconfigs/t/TN93/TN93-1.0.7-GCCcore-10.3.0.eb [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+
+buildininstalldir = True
```
Comment (smoors) [inline easybuild/easyconfigs/t/TN93/TN93-1.0.7-GCCcore-10.3.0.eb:25]: I don't see any reason why you would do this, as you are copying everything into the correct installation directory anyway.

File: easybuild/easyconfigs/t/TN93/TN93-1.0.7-GCCcore-10.3.0.eb [hunk 11]
```diff
@@ -0,0 +1,50 @@
...
+
+builddependencies = [
+    ('CMake', '3.20.1'),
+    ('binutils', '2.36.1'),
+]
+
+build_cmd = ' make '
```
Comment (smoors) [inline easybuild/easyconfigs/t/TN93/TN93-1.0.7-GCCcore-10.3.0.eb:32]: please remove this: the default build command is `make -j x`. you can change the x with parameter `parallel` if needed.

**Review by smoors — CHANGES_REQUESTED** [review]

**Comment by smoors** [comment]
@boegelbot: please test @ generoso

**Comment by smoors** [comment]
Test report by @smoors
**SUCCESS**
Build succeeded for 4 out of 4 (2 easyconfigs in total)
node300.hydra.os - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) Gold 6148 CPU @ 2.40GHz (skylake_avx512), Python 2.7.5
See https://gist.github.com/b934b7a3b360a28eb29e73b059887a73 for a full test report.

**Review by smoors — APPROVED** [review]
lgtm

**Comment by smoors** [comment]
Going in, thanks @i-mtz!

**Pull request merged**